src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -102,8 +128,21 @@ private void CompilationStartAction(CompilationStartAnalysisContext context)
             var builder = ImmutableDictionary.CreateBuilder<string, string>();
             builder[nameof(NamingStyle)] = applicableRule.NamingStyle.CreateXElement().ToString();
             builder["OptionName"] = nameof(SimplificationOptions.NamingPreferences);
-            builder["OptionLanguage"] = context.Compilation.Language;
-            context.ReportDiagnostic(Diagnostic.Create(descriptor, context.Symbol.Locations.First(), builder.ToImmutable()));
+            builder["OptionLanguage"] = compilation.Language;
+
+            return Diagnostic.Create(descriptor, symbol.Locations.First(), builder.ToImmutable());
+
+            async Task<NamingStylePreferences> GetNamingStylePreferencesAsync()
```

Comment:
``// local functions.

Reply:
that's insane.. there should be a warning comment before every single local function?

Reply:
No.  Just before a **group** of them.  It makes PRs much easier to understand as it's more obvious that these are local functions and thus may capture, and also taht they don't have to have things like accessibility.

Reply:
fixed